src/utils/chat.ts

```diff
@@ -60,26 +32,29 @@ export const getPreviewChatMessage = ({
   counterPartyUser: user,
 });
 
-export interface GetPreviewChatMessageToDisplayProps {
-  message: PreviewChatMessage;
-  myUserId?: string;
-}
-
-export const getPreviewChatMessageToDisplay = ({
-  message,
-  myUserId,
-}: GetPreviewChatMessageToDisplayProps): PreviewChatMessageToDisplay => ({
-  ...message,
-  isMine: myUserId === message.from,
-});
-
 export const buildMessage = <T extends ChatMessage>(
-  message: Pick<T, Exclude<keyof T, "ts_utc">>
+  from: WithId<ChatUser>,
+  message: Pick<T, Exclude<keyof T, "ts_utc" | "from">>
 ) => ({
   ...message,
+  from: pickChatUserFromUser(from),
   ts_utc: firebase.firestore.Timestamp.now(),
 });
 
+export const pickChatUserFromUser = (user: WithId<User>) =>
+  pick(user, "id", "partyName", "pictureUrl", "anonMode") as WithId<ChatUser>;
```

Comment:
```suggestion
export const pickChatUserFromUser = (user: WithId<User>): WithId<ChatUser> =>
  pick(user, "id", "partyName", "pictureUrl", "anonMode");
```

We should really avoid usage of `as`